ToolChain/Assembler/Assembler.py

```diff
@@ -56,79 +50,90 @@ def __init__(self, inputFile=None, outputFile=None):
-        fileContent += b"</ExternalSymbols>"
-
-        # Deal with the text section
-        fileContent += b"<Text>"
-        for instruction in self.parser.instructionList:
-            fileContent += instruction
-        fileContent += b"</Text>"
-
-        return fileContent
-
-    def writeAssembledFile(self, file=None, outputData=None):
-        """
-        This will write the assembled file to disk
-        :param file: str, output file path
-        :param outputData: bytes, the data to be written on the disk
+        This method assembles the binary data itself. It will parse each line individually and extract the info needed:
+        1.If the label flag is set to 0, it's a regular instruction, string, number, of memref.
+        2.If the label flag is set to 1, it's an internal symbol.
+        3.If the label flag is set to 2, it's an external symbol.
+        In cases 2 and 3, the symbols are added to their respective lists along with their memory location.
+        Memory locations are relative to the beginning of the file. Instructions and data are given set lengths.
+        Internal and external symbol lists contain tuples containing the symbol name, and its offset.
+        These will be used to fill the .o file information before the binary data.
+        :param inputFile: str, the name of the file to be read.
+        :param outputFile: str, the name of the file to be written to.
         :return:
         """
 
-        file = open(file, mode="wb")
-        file.write(outputData)
+        parser = Parser()                    # Parser object which will return the binary code for instructions
+        masterString = b""                   # String that will eventually be our output file
+        xmlstring = b""                      # String that will contain the "xml" data at the beginning of the .o file
+        globalList = []                      # Temporary list to hold the names of global(external) symbols
+        externalList = []                    # Master list of all external symbols to be used by linker
+        internalList = []                    # Master list of all internal symbols
+        offset = 0                           # Offset used to calculate a label's offset from the beginning of the file
+        lineno = 0                           # Keeps track of which line we're parsing to pinpoint errors
+
+        instructionFlag = 0                  # Flag for an operand, meaning parser found a valid instruction
+        localReferenceFlag = 1               # Flag for local function, internal symbol
+        globalReferenceFlag = 2              # Flag for external function, global symbol
+        commentFlag = 3                      # Flag for comment
+
+        file = open(inputFile, mode="r")
+        fileLines = file.readlines()
         file.close()
+
+        masterString += b"<Text>"
+
+        for line in fileLines:
+            lineno += 1
+            # If the file isn't empty, process the content
```

Comment:
Too many blank lines. This breaks the code "structure".

Reply:
The whole inner part of this for loop could be reworked to improve code readability.